Add an easy way and examples of activating objects for a single time step

GiudGiud opened the pull request:
refs #31751 #31752

GiudGiud committed:
Add some debugging output when failing to find the right chain control name
refs #31752

GiudGiud committed:
Add ability to trip on old values
refs #31751

GiudGiud committed:
Add tests for chain control activation of objects with common needs
refs #31751

GiudGiud committed:
Add debug output of the chain control data
refs #31752

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3309445/), step Docs: sync website on 99dc447 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31753/site)

This comment will be updated on new commits.

lindsayad commented on framework/src/base/ChainControlDataSystem.C:
```diff
@@ -29,3 +29,21 @@ ChainControlDataSystem::getChainControlDataMap() const
+    if (const auto ctl_real = dynamic_cast<ChainControlData<Real> *>(item.second.get()))
+      map_str += ". Current value: " + std::to_string(ctl_real->get());
+    if (const auto ctl_bool = dynamic_cast<ChainControlData<bool> *>(item.second.get()))
+      map_str += ". Current value: " + std::to_string(ctl_bool->get());
```
Can these be something other than `Real` or `bool`? If so, we should handle it or error or do something to let the user know what's going on

lindsayad commented on framework/src/problems/SubProblem.C:
```diff
@@ -1216,11 +1222,13 @@ SubProblem::jacobianSetup()
     showFunctorRequestors();
   }
+  if (_show_chain_control_data)
+    _console << _app.getChainControlDataSystem().outputChainControlMap();
```
```suggestion
    _console << _app.getChainControlDataSystem().outputChainControlMap() << std::flush;
```
Or don't flush above. Let's be consistent

GiudGiud replied:
will update

GiudGiud committed:
Address Alex's review:
- plan for future control data types in th debug output
- consistent output in subproblem

GiudGiud commented:
Thanks for the review!
